lib-extra/src/main/java/com/diffplug/spotless/extra/groovy/GrEclipseFormatterStep.java

```diff
@@ -16,76 +16,82 @@
 package com.diffplug.spotless.extra.groovy;
 
 import java.io.File;
-import java.io.Serializable;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.Objects;
 import java.util.Properties;
 
-import com.diffplug.spotless.FileSignature;
 import com.diffplug.spotless.FormatterFunc;
-import com.diffplug.spotless.FormatterProperties;
 import com.diffplug.spotless.FormatterStep;
-import com.diffplug.spotless.JarState;
 import com.diffplug.spotless.Provisioner;
+import com.diffplug.spotless.ThrowingEx;
+import com.diffplug.spotless.extra.config.EclipseConfiguration;
+import com.diffplug.spotless.extra.config.EclipseConfiguration.State;
 
 /** Formatter step which calls out to the Groovy-Eclipse formatter. */
-public class GrEclipseFormatterStep {
-	private static final String NAME = "groovy-eclipse formatter";
+public class GrEclipseFormatterStep implements ThrowingEx.Function<EclipseConfiguration.State, FormatterFunc> {
+	private static final String NAME = "groovy eclipse formatter";
 	private static final String FORMATTER_CLASS = "com.diffplug.gradle.spotless.groovy.eclipse.GrEclipseFormatterStepImpl";
-	private static final String DEFAULT_VERSION = "2.3.0";
-	private static final String MAVEN_COORDINATE = "com.diffplug.spotless:spotless-ext-greclipse:";
 	private static final String FORMATTER_METHOD = "format";
+	static final String VERSIONS[] = {"2.3.0", "4.6.3"};
 
-	/** Creates a formatter step using the default version for the given settings file. */
+	/**
+	 * Creates a formatter step using the default version for the given settings file.
+	 */
+	@Deprecated
 	public static FormatterStep create(Iterable<File> settingsFiles, Provisioner provisioner) {
 		return create(defaultVersion(), settingsFiles, provisioner);
 	}
 
 	/** Creates a formatter step for the given version and settings file. */
+	@Deprecated
 	public static FormatterStep create(String version, Iterable<File> settingsFiles, Provisioner provisioner) {
-		return FormatterStep.createLazy(
-				NAME,
-				() -> new State(JarState.from(MAVEN_COORDINATE + version, provisioner), settingsFiles),
-				State::createFormat);
+		EclipseConfiguration config = createConfig(provisioner);
+		config.setVersion(version);
+		config.setPreferences(settingsFiles);
+		return createStep(config);
 	}
 
+	@Deprecated
 	public static String defaultVersion() {
-		return DEFAULT_VERSION;
+		return "2.3.0";
```

Comment:
Maybe `VERSIONS[0]` or `VERSIONS[VERSIONS.length-1]`?

Reply:
Currently the SemanticVersions are sorted by the EclipseConfiguration/EclipseBasedStepBuilder. I did not want to implicitly force the step implementation to do that. The return value "2.3.0" is in a way correct. The method is deprecated. You get the version you always got with prior versions for extra-lib.

Reply:
👍